app/src/main/java/com/github/geohunt/app/ui/components/bounties/AdminBountyPage.kt

```diff
@@ -0,0 +1,248 @@
+            bounty = bountyState.value!!,
+            teams = teamsState.value,
+            challenges = challengesState.value,
+            setName = viewModel::setBountyName,
+            onCreateChallenge = onCreateChallenge
+        )
+    } else {
+        CircularProgressIndicator()
+    }
+}
+
+@Composable
+private fun AdminBountyPageUI(
+    bounty: Bounty,
+    setName: (String) -> Unit,
+    teams: List<Team>,
+    challenges: List<Challenge>,
+    onCreateChallenge: () -> Unit,
+) {
+    val memberCount = teams.sumOf { it.membersUid.size }
+    var showPopup by remember { mutableStateOf(false) }
+
+    if (showPopup) {
+        RenameBountyPopup(bounty) { newNameOpt ->
+            showPopup = false
+            newNameOpt?.apply { setName(this) }
+        }
+    }
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .verticalScroll(rememberScrollState())
+            .padding(5.dp, 10.dp)
+            .testTag("admin-bounty-page-loaded")
+    ) {
+        Row(modifier = Modifier.padding(10.dp, 0.dp)) {
+            Text(
+                text = bounty.name,
+                fontSize = 28.sp
+            )
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            IconButton(onClick = { showPopup = true },
+                modifier = Modifier
+                    .testTag("edit-btn")
+                    .align(Alignment.CenterVertically)) {
+                Icon(Icons.Default.Edit,
+                    contentDescription = "edit")
+            }
+
+            Spacer(modifier = Modifier.width(5.dp))
+
+            Icon(Icons.Default.AdminPanelSettings,
+                modifier = Modifier.align(Alignment.CenterVertically),
+                contentDescription = "admin"
+            )
+        }
+
+        Row(modifier = Modifier.padding(20.dp, 0.dp)) {
+            val firstComponent = formatTimeStuff(start = bounty.startingDate, end = bounty.expirationDate)
+            val secondComponent = pluralStringResource(id = R.plurals.teams_count, count = teams.size, teams.size)
+            val thirdComponent = pluralStringResource(id = R.plurals.members_count, count = memberCount, memberCount)
+
+            Text(
+                text = "$firstComponent - $secondComponent - $thirdComponent",
+                fontSize = 11.sp,
+                color = MaterialTheme.colors.primaryVariant
+            )
+
+            Spacer(modifier = Modifier.weight(1f))
+
+            Text(
+                text = DateFormatUtils.formatRange(bounty.startingDate, bounty.expirationDate),
+                fontSize = 11.sp,
+                fontStyle = FontStyle.Italic,
+                color = MaterialTheme.colors.primaryVariant
+            )
+        }
+
+        HorizontalDivider(10.dp)
+
+        DisplayTeams(teams)
+
+        Spacer(modifier = Modifier.height(10.dp))
+
+        DisplayChallenges(challenges, onCreateChallenge)
+    }
+}
+
+@Composable
+private fun ColumnScope.DisplayTeams(

```

Comment:
Why do we need a leaderboard since we already have one ? Or maybe there should be some admin rights to delete teams or whatever